reactive/kubernetes_master.py

```diff
@@ -1974,6 +2020,38 @@ def configure_apiserver():
                                  api_opts, 'api-extra-args')
     service_restart('snap.kube-apiserver.daemon')
 
+    is_leader = is_state('leadership.is_leader')
+    if is_leader and is_service_cidr_expansion():
```

Comment:
Can you update this if condition (or update `is_service_cidr_expansion`) to also check that the service-cidr has changed?

As it is now, it looks like if the service-cidr hasn't changed, then `is_subnet_of(current_service_cidr, new_service_cidr)` will return True, therefore `is_service_cidr_expansion()` will return True, therefore this condition will be True on the leader.

As a result, every time the charm reconfigures kube-apiserver, even if service-cidr hasn't changed, it will delete the kubernetes service and do a rolling restart of addons. That's a bit too disruptive for us to be doing it when we don't need to.

Reply:
Ah, good catch.  However, checking in either place will not work without additional changes since the new CIDR is already frozen by the time we get here.

```suggestion
    if was_service_cidr_expanded and is_state('leadership.is_leader'):
```